framework/include/interfaces/MortarInterface.h

```diff
@@ -0,0 +1,76 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+
+#include <set>
+
+class InputParameters;
+class MooseObject;
+class FEProblemBase;
+class MooseMesh;
+class MortarData;
+
+/**
+ * An interface for accessing mortar mesh data
+ *
+ * \note moi is shorthand for mortar interface, so \p _moi_problem indicates
+ *       the mortar interface's problem
+ */
+class MortarInterface
+{
+public:
+  MortarInterface(const MooseObject * moose_object);
+
+  static InputParameters validParams();
+
+  /**
+   * Returns the master lower dimensional subdomain id
+   */
+  SubdomainID masterSubdomain() const { return _master_subdomain_id; }
```

Comment:
Need slaveSubdomain as well?

Reply:
The only reason this API exists is for subdomain coverage checks in the `ConstraintWarehouse`. I'm able to get the slave subdomain through the associated lagrange multiplier variable, but because it doesn't live on the master subdomain I had to add this API to `MortarData` to get the master